presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateTDigestDoublePercentileAggregations.java

```diff
@@ -0,0 +1,138 @@
+        DOUBLE.writeDouble(blockBuilder, percentile);
+        return blockBuilder.build();
+    }
+
+    @InputFunction
+    public static void weightedInput(@AggregationState TDigestAndPercentileState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.BIGINT) long weight, @SqlType(StandardTypes.DOUBLE) double percentile)
+    {
+        checkCondition(weight > 0, INVALID_FUNCTION_ARGUMENT, "Percentile weight must be positive");
+
+        initializePercentilesArray(state, createPercentileBlock(percentile));
+        initializeDigest(state, STANDARD_COMPRESSION_FACTOR);
+
+        TDigest digest = state.getDigest();
+        state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+        digest.add(value, weight);
+        state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+
+        // use last percentile
+        state.setPercentiles(ImmutableList.of(percentile));
+    }
+
+    @InputFunction
+    public static void weightedInput(@AggregationState TDigestAndPercentileState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.BIGINT) long weight, @SqlType(StandardTypes.DOUBLE) double percentile, @SqlType(StandardTypes.DOUBLE) double compression)
+    {
+        checkCondition(weight > 0, INVALID_FUNCTION_ARGUMENT, "Percentile weight must be positive");
+
+        initializePercentilesArray(state, createPercentileBlock(percentile));
+        initializeDigest(state, compression);
+
+        TDigest digest = state.getDigest();
+        state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+        digest.add(value, weight);
+        state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+
+        // use last percentile
+        state.setPercentiles(ImmutableList.of(percentile));
+    }
+
+    @CombineFunction
+    public static void combine(@AggregationState TDigestAndPercentileState state, TDigestAndPercentileState otherState)
+    {
+        TDigest input = otherState.getDigest();
+        TDigest previous = state.getDigest();
+
+        if (previous == null) {
+            state.setDigest(input);
+            state.addMemoryUsage(input.estimatedInMemorySizeInBytes());
+        }
+        else {
+            state.addMemoryUsage(-previous.estimatedInMemorySizeInBytes());
+            previous.merge(input);
+            state.addMemoryUsage(previous.estimatedInMemorySizeInBytes());
+        }
+
+        state.setPercentiles(otherState.getPercentiles());
+    }
+
+    @OutputFunction(StandardTypes.DOUBLE)
+    public static void output(@AggregationState TDigestAndPercentileState state, BlockBuilder out)
+    {
+        TDigest digest = state.getDigest();
+        List<Double> percentiles = state.getPercentiles();
+
+        if (percentiles == null || digest == null) {
+            out.appendNull();
+            return;
+        }
+
+        checkState(percentiles.get(0) != -1.0, "Percentile is missing");
```

Comment:
When will this happen?

Reply:
It should never happen, as the default value for a numeric column in SQL is 0. Kept check from `ApproximateLongPercentileAggregation`.